test/commands/generate-cr.test.ts

```diff
@@ -335,4 +330,56 @@ describe('generate-cr', () => {
       expect((error as CLIError).oclif.exit).eq(2)
     })
     .it("generate-cr -f can't be used without -o")
+
+
+  // TAG FILTERING TESTS
+  executeTagFilteringTest(['dev'], devTags);
+  executeTagFilteringTest(['prod'], prodTags);
+  executeTagFilteringTest(['dev', 'prod'], [...devTags, ...prodTags]);
+
+  function executeTagFilteringTest(tagTypes: string[], expectedTags: string[]) {
+
+    test
+      .stdout()
+      .stderr()
+      .do(() => {
+        sinon.stub(DockerService, 'listTags').resolves(allTags)
+        const stubDigestsExecutor = createStubDigestsExecutor();
+        sinon
+          .stub(DockerService, 'getDigestsExecutor')
+          .returns(stubDigestsExecutor)
+        sinon.stub(DockerService, 'getYamlDescriptorFromImage').resolves(yamlDescriptor)
+      })
+      .stub(CliUx.ux, 'confirm', () => sinon.stub().resolves(true))
+      .command([
+        'generate-cr',
+        '-i',
+        'my-org/my-image',
+        '-d',
+        '-t',
+       ...tagTypes
+      ])
+      .it(
+        `Filter tags by ${tagTypes} tag type`,
+        () => {
+          const listTagsStub = DockerService.getDigestsExecutor as sinon.SinonStub
+          sinon.assert.calledWith(
+            listTagsStub,
+            'registry.hub.docker.com/my-org/my-image',
+            expectedTags)
+        }
+      )
+  }
 })
+
+function createStubDigestsExecutor() {
+  return new (class extends StubParallelProcessExecutorService {
+    async getDigests(): Promise<Map<string, string>> {
+      await super.execute()
+      const tagsWithDigests = new Map()
+      tagsWithDigests.set('0.0.2', 'sha256:abcd')
+      tagsWithDigests.set('0.0.1', 'sha256:1234')
+      return tagsWithDigests
+    }
+  })([0, 0])
```

Comment:
To avoid certain random issues we experience in the past with events in tests the number of stubbed results passed to the constructor should be the same of the expected progress bar steps.
I suggest passing a parameter like `resultsCount: number` to the `createStubDigestsExecutor()` function and use something like `new Array(resultsCount).fill(0)` instead of (`[0,0]`) here.